src/edu/csus/ecs/pc2/core/ParseArguments.java

```diff
@@ -471,6 +499,27 @@ public void setRequireArgOpts(String newRequireArgOpts) {
         requireArgOpts[0] = newRequireArgOpts;
     }
 
+    /**
+     * Set options which are allowed
+     *
+     * @param newAllowedOpts
+     *            java.lang.String[]
+     */
+    public void setAllowedOpts(java.lang.String[] newAllowedOpts) {
+        allowedOptions = newAllowedOpts;
+    }
+
+    /**
+     * Set a single allowed option.
+     *
+     * @param newAllowedOpt
+     *            java.lang.String
+     */
+    public void setAllowedOpt(String newAllowedOpt) {
+        allowedOptions = new String[1];
+        allowedOptions[0] = newAllowedOpt;
+    }
+
```

Comment:
Unless I'm reading things wrong, if someone first calls `setAllowedOpts(String [])`, and then later makes a call to `setAllowedOpt(String)`, the effect would be to _wipe out_ all the options which were set by the original call to `setAllowedOpts(String [])`.  I would not expect that to be desired/correct behavior; if I set a collection of options with a call to `setAllowedOpts(String [])` and then call `setAllowedOpt(String)`, I would expect the new "allowed option" to be ADDED to the set of (previously-defined) allowed options.  

It seems to me that method `setAllowedOpt(String)` should check for a non-null `allowedOptions` and if present it should ADD the specified option to the existing list.

Reply:
For consistency, I made the new methods work the same way that "RequiredArgOpts" methods work.  That is, either "set" method (String[]) or (String) creates a completely new array.  I would expect if you wanted to append an additional option that's allowed, there would be an "addAllowOpt()" method.

Reply:
Ok.  I could argue that the design of the original methods, from which you say these are copied, is also flawed in the same way... but let's not go down that rabbit hole.

I do think it would be helpful if the methods (at least the new ones) had a comment in their header that at least made it clear that calling either one of them has the side-effect of throwing away any previously-set options.  That way at least future developers have some documentation that warns them...